### PR TITLE
Add new testcases

### PR DESCRIPTION
**Pull request by nukenukenukelol**
testRayInterSectThetaBoundaryOutsideTheCircle
testRayDoesNotGoThroughTheInnermostVoxel
testRayTraverseSingleVoxel
testRayEndsInsideCircle
testRayWithPositiveXNegativeY
testRayWithNegativeXPositiveY
testRayBeginTimeDiffThan0
testRayWithOddNumAngularSections
testRaySlightIntersect

### CODE REVIEW

File: polarCoordinateTraversalTest.m [hunk 1]
```diff
@@ -278,15 +278,211 @@ function testRayGoesAlongThetaBoundary(testCase)
         verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
 end
 
-% TODOs
 % - Ray intersects a “theta boundary” outside the circle, and ensure it does 
 %   NOT record this as a theta hit.
+function testRayInterSectThetaBoundaryOutsideTheCircle(testCase)
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: ```testRayIntersect...```
```Intersect``` is one word, so no need to capitalize to ```InterSect```.

Comment (nukenukenukelol) [reply polarCoordinateTraversalTest.m]: Ooof aight let me fix that

File: polarCoordinateTraversalTest.m [hunk 2]
```diff
@@ -278,15 +278,211 @@ function testRayGoesAlongThetaBoundary(testCase)
...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [2,2,1];
+        expected_thetaVoxels = [1,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
 % - Ray that has an odd number of angular sections.
+function testRayWithOddNumAngularSections(testCase):
```
Comment (cgyurgyik) [inline polarCoordinateTraversalTest.m]: This doesn't have an odd number of angular sections, and it looks like it doesn't traverse any theta voxels

Comment (nukenukenukelol) [reply polarCoordinateTraversalTest.m]: Actually I'm a little bit confused. When you say there's an odd number of angular sections, what do you mean? Like the ray intersect angular sections with odd number of times?

Comment (cgyurgyik) [reply polarCoordinateTraversalTest.m]: the ```num_angular_sections``` is not even, i.e. ```num_angular_sections = 5;```

**Pull request merged**